### PR TITLE
Move Url decoding/encoding to Uri companion object

### PR DESCRIPTION
**Pull request by IndiscriminateCoding**

### CODE REVIEW

**Comment by IndiscriminateCoding** [comment]
Binary compatibility checks are failed due to missing `UrlCodingUtils` object.
Is it OK (it was package-private) or should I make a PR against **master**?

**Comment by rossabaker** [comment]
It is okay to add exceptions for package privates.

File: core/src/main/scala/org/http4s/Uri.scala [hunk 1]
```diff
@@ -374,6 +377,129 @@ object Uri {
...
+    *
+    * @param toEncode the string to encode
+    * @param charset the charset to use for characters that are percent encoded
+    * @param spaceIsPlus if space is not skipped, determines whether it will be
+    * rendreed as a `"+"` or a percent-encoding according to `charset`.
+    * @param toSkip a predicate of characters exempt from encoding.  In typical
+    * use, this is composed of all Unreserved URI characters and sometimes a
+    * subset of Reserved URI characters.
+    */
+  def urlEncode(
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/Uri.scala]: Bikeshed: why (other than private historical reasons) `url` in a class called `Uri`?  Maybe just `encode`?  Similar for `decode`.

Comment (IndiscriminateCoding) [reply core/src/main/scala/org/http4s/Uri.scala]: You are right! Done.

**Comment by rossabaker** [comment]
Looks like this just needs MiMa exceptions.

**Pull request merged**